tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -136,6 +136,13 @@ public void CloseSpan(Span span)
                     _iastRequestContext.AddIastVulnerabilitiesToSpan(span);
                     OverheadController.Instance.ReleaseRequest();
                 }
+                else
+                {
+                    if (Iast.Iast.Instance.Settings.Enabled)
```

Comment:
The "overhead controller" mentioned in the description sets `_iastRequestContext` to null when the overhead limit is reached?

Minor: I find checking `Iast.Iast.Instance.Settings.Enabled` and then setting `AddIastDisabledToSpan` a bit confusing, it looks like a mistake. Perhaps you could change the if just to check `_iastRequestContext` on the outer if.

Reply:
_iastRequestContext is only initialized when the overhead controller allows a request to be analyzed. This is done for resource optimization, by default it is null.

I will can change the "if" condition to achieve a better readability.

Reply:
Done!